Shoko.Server/API/v3/Controllers/PlexController.cs

```
@@ -0,0 +1,203 @@
+            return BadRequest(AlreadyAuthenticated);
+
+        return helper.LoginUrl;
+    }
+
+    /// <summary>
+    /// Check if the current user is currently authenticated against the plex api.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("Auth/IsAuthenticated")]
+    public ActionResult<bool> IsAuthenticated()
+    {
+        var helper = GetHelperForUser();
+
+        return helper.IsAuthenticated;
+    }
+
+    /// <summary>
+    /// Invalidate and remove the current plex authentication token.
+    /// </summary>
+    /// <returns></returns>
+    [HttpPost("Auth/InvalidateToken")]
+    public ActionResult<bool> InvalidateToken()
+    {
+        var helper = GetHelperForUser();
+
+        if (!helper.IsAuthenticated)
+            return false;
+
+        helper.InvalidateToken();
+
+        return true;
+    }
+
+    /// <summary>
+    /// Show all available server for the current user.
+    /// </summary>
+    /// <param name="userId"></param>
+    /// <returns></returns>
+    [HttpGet("Linking/AvailableServers")]
+    public ActionResult<List<PlexServer>> AvailableServers(int userId)
+    {
+        var helper = GetHelperForUser();
+
+        if (!helper.IsAuthenticated)
+            return BadRequest(NotAuthenticated);
+
+        var currentServerID = helper.ServerCache?.ClientIdentifier ?? "";
+        return helper.GetPlexServers()
+            .Select(device => new PlexServer(device, device.ClientIdentifier == currentServerID))
+            .ToList();
+    }
+
+    /// <summary>
+    /// Get the active server.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("Linking/Server")]
+    public ActionResult<PlexServer> CurrentDevice()
+    {
+        var helper = GetHelperForUser();
+
+        if (!helper.IsAuthenticated)
+            return BadRequest(NotAuthenticated);
+
+        var currentServer = helper.ServerCache;
+        if (currentServer == null)
+            return BadRequest(NoSelectedServer);
```

Should an "unlinked" be a bad request? I would expect this to be a http 204

---

That's the difference between loudly complaining (this 400) and silently failing (204). Personally I want it to loudly complain, but silently failing (and returning an empty) body is also fine. Will update it to use 204 instead.

---

Your still returning a 200 series response, aka a successful request, 400 indicates something was wrongly requested on the client & using 204 or “No content” is still distinct enough if you want to determine based off the return code